ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClient.java

```diff
@@ -57,87 +60,76 @@ public Set<Relationship> getRelationships() {
+		boolean timestampPath = streamTimestampPathProperty(context);
+		boolean versionOfPath = streamVersionOfProperty(context);
+		boolean shape = streamShapeProperty(context);
+		ldesProperties = new LdesPropertiesExtractor(requestExecutor).getLdesProperties(ldes, timestampPath,
+				versionOfPath, shape);
 	}
 
 	@Override
 	public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
-		if (ldesService.hasFragmentsToProcess()) {
-			Lang dataDestinationFormat = LdesProcessorProperties.getDataDestinationFormat(context);
-			LdesFragment fragment = ldesService.processNextFragment();
-
-			String timestampPath = streamTimestampPathProperty(context) ? fragment.getTimestampPath()
-					.orElseThrow(() -> new LdesPropertyNotFoundException(LdesFragment.LDES_TIMESTAMP_PATH.toString()))
-					: null;
-			String versionOfPath = streamVersionOfProperty(context) ? fragment.getVersionOfPath()
-					.orElseThrow(() -> new LdesPropertyNotFoundException(LdesFragment.LDES_VERSION_OF.toString()))
-					: null;
-			String shape = streamShapeProperty(context) ? fragment.getShaclShape()
-					.orElseThrow(() -> new LdesPropertyNotFoundException(LdesFragment.TREE_SHAPE.toString()))
-					: null;
-
-			// Send the processed members to the next Nifi processor
-			fragment.getMembers()
-					.forEach(ldesMember -> {
-						FlowFile flowFile = session.create();
-
-						if (streamTimestampPathProperty(context)) {
-							session.putAttribute(flowFile, "ldes.timestamppath", timestampPath);
-						}
-						if (streamVersionOfProperty(context)) {
-							session.putAttribute(flowFile, "ldes.isversionofpath", versionOfPath);
-						}
-						if (streamShapeProperty(context)) {
-							session.putAttribute(flowFile, "ldes.shacleshapes", shape);
-						}
-
-						FlowManager.sendRDFToRelation(session, flowFile,
-								ModelConverter.convertModelToString(ldesMember.getMemberModel(), dataDestinationFormat),
-								DATA_RELATIONSHIP, dataDestinationFormat);
-					});
+		SuppliedMember memberRecord = memberSupplier.get();
+
+		FlowFile flowFile = session.create();
+
+		if (streamTimestampPathProperty(context)) {
+			session.putAttribute(flowFile, "ldes.timestamppath", ldesProperties.getTimestampPath());
+		}
+		if (streamVersionOfProperty(context)) {
+			session.putAttribute(flowFile, "ldes.isversionofpath", ldesProperties.getVersionOfPath());
+		}
+		if (streamShapeProperty(context)) {
+			session.putAttribute(flowFile, "ldes.shacleshapes", ldesProperties.getShape());
 		}
+		Lang dataDestinationFormat = LdesProcessorProperties.getDataDestinationFormat(context);
+		FlowManager.sendRDFToRelation(session, flowFile,
+				convertModelToString(memberRecord.getModel(), dataDestinationFormat),
+				DATA_RELATIONSHIP, dataDestinationFormat);
 	}
 
 	@OnRemoved
 	public void onRemoved() {
-		ldesService.getStateManager().destroyState();
+		treeNodeProcessor.destoryState();
```

Comment:
can we add this api to the memberSupplier? Then we only need to interact with that one interface, minimizing the required knowledge of how the client core works